src/Client/AuthorityUrlValidationStrategy.cs

```diff
@@ -0,0 +1,62 @@
+﻿using IdentityModel.Internal;
+using System;
+using System.Collections.Generic;
+
+namespace IdentityModel.Client
+{
+    /// <summary>
+    /// Implementation of <see cref="IAuthorityValidationStrategy"/> based on <see cref="Uri"/> equality.
+    /// </summary>
+    public sealed class AuthorityUrlValidationStrategy : IAuthorityValidationStrategy
+    {
+        /// <summary>
+        /// Singleton instance.
+        /// </summary>
+        public static AuthorityUrlValidationStrategy Instance = new AuthorityUrlValidationStrategy();
```

Comment:
Is there a good reason to use the "Instance" pattern here?

Reply:
My reasoning about that is that creating different instances brings no benefit over singleton "Instance" pattern (no parameterization and no state). Probably not a "good" reason. I removed that.